src/docs/sdk/performance/opentelemetry.mdx

```diff
@@ -18,7 +18,220 @@ When Sentry performance monitoring was initially introduced, OpenTelemetry was i
 
 ## Approach
 
-TODO: Talk about the approach we are using, based on Matt's hackweek project - https://github.com/getsentry/sentry-ruby/pull/1876
+To transform OpenTelemetry SDK data to Sentry data, we rely on the [OpenTelemetry SpanProcessor](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#span-processor). The SpanProcessor is a component that runs on the same thread as the application code, so it effectively runs in process alongside user code and the Sentry SDK.
+
+### Step 1: Implement the SentrySpanProcessor on your SDK
+
+Below we have written a custom SpanProcessor that will transform the OpenTelemetry spans to Sentry spans.
+
+This relies on a couple key assumptions.
+
+1. Hub/Scope propogation works properly in your platform. This means that the hub used in `onStart` is the same hub used in `onEnd` and it means that hubs fork properly in async contexts.
+2. The SDK is initialized before the OpenTelemetry SDK is initialized.
+3. There will only be a single transaction occuring at the same time. This is a limitation of the current SDK design.
+
+```ts
+import { SpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { Span as OpenTelemetrySpan } from '@opentelemetry/api';
+import { Span as SentrySpan } from '@sentry/tracing';
+
+class SentrySpanProcessor implements SpanProcessor {
+  /**
+   * A map that stores a spanId to a tuple of a span and it's parent span.
+   * Used to decide what span gets finished.
+   */
+  private readonly _map: Record<SentrySpan['spanId'], [span: SentrySpan, parentSpan?: SentrySpan]>;
+
+  constructor(options: SentrySpanProcessorOptions) {
+    this._options = options;
+  }
+
+  onStart(otelSpan: OpenTelemetrySpan): void {
+    const hub = Sentry.getCurrentHub();
+    const scope = hub.getScope();
+
+    if (!hub || !scope) return;
+
+    const otelSpanId = span.spanContext().spanId;
+
+    const parentSpan = scope.getSpan();
+    if (parentSpan) {
+      // If the parentSpan exists, we create a new span as a child of the parentSpan
+      const childSpan = parentSpan.startChild({
+        description: otelSpan.name,
+        instrumenter: 'otel',
+        // don't define op because we will do that on span.finish
+      });
+
+      if (childSpan) {
```

Comment:
Can `parentSpan.startChild()` return null/undefined?

Reply:
no it can't, actually co-pilot is the one that made this, I can clean this up.